src/app/components/Profile/Profile.tsx

```diff
@@ -0,0 +1,82 @@
+import { useEffect, useState } from 'react';
+import defaultAvatar from './defaultAvatar.svg';
+import classes from './Profile.module.css';
+import { Profile, getCurrentUser } from '../../utils/user';
+
+const defaultProfile: Profile = {
+  username: 'Game-Ticker',
+  displayName: 'Game-Ticker',
+  avatar: null,
+};
+
+function openLoginDialog() {
+  overwolf.profile.openLoginDialog();
+}
+
+const User = (): JSX.Element => {
+  const [profile, setProfile] = useState<Profile>(defaultProfile);
+  const [profileError, setProfileError] = useState<string | null>(null);
+  const [loginDialog, setLoginDialog] = useState(false);
+  const [login, setLogin] = useState(false);
+
+  useEffect(() => {
+    overwolf.profile.onLoginStateChanged.addListener((result) => {
+      if (result.connectionState !== 'Online') {
+        setLogin(false);
+        return;
+      }
+      setLogin(true);
+    });
+    const getUser = async () => {
```

Comment:
It sounds like a function which returns a user, but it returns nothing. Maybe rename to `loadUser` or `refreshUser`?